go.mod

```diff
@@ -56,7 +56,6 @@ require (
 	github.com/lithammer/dedent v1.1.0
 	github.com/lpabon/godbc v0.1.1 // indirect
 	github.com/moby/ipvs v1.0.1
-	github.com/moby/sys/mountinfo v0.6.0 // indirect
```

Comment:
Hmm, this hunk does not look good. It should be something like
```diff
-       github.com/moby/sys/mountinfo v0.6.0 // indirect
+       github.com/moby/sys/mountinfo v0.6.1 // indirect
```

I have checked vendoring tools and they work fine.

What you need to create this commit is
```console
$ ./hack/pin-dependency.sh github.com/moby/sys/mountinfo v0.6.1
$ ./hack/update-vendor.sh
$ ./hack/lint-dependencies.sh 
```

(and the usual `git` commands)

Reply:
I did exactly what you told.

```
  kubernetes git:(improve-mount-detection) ./hack/pin-dependency.sh github.com/moby/sys/mountinfo v0.6.1
Running: go mod download github.com/moby/sys/mountinfo@v0.6.1
Resolved to github.com/moby/sys/mountinfo@v0.6.1
Running: go mod edit -require github.com/moby/sys/mountinfo@v0.6.1
Running: go mod edit -replace github.com/moby/sys/mountinfo=github.com/moby/sys/mountinfo@v0.6.1

Run hack/update-vendor.sh to rebuild the vendor directory
➜  kubernetes git:(improve-mount-detection) ✗ ./hack/update-vendor.sh                                      
+++ [0425 17:40:58] logfile at /tmp/update-vendor.QxrQ/update-vendor.log
+++ [0425 17:40:58] go.mod: update staging references
+++ [0425 17:41:03] go.mod: propagate to staging modules
+++ [0425 17:41:03] go.mod: sorting staging modules
+++ [0425 17:41:18] go.mod: tidying
+++ [0425 17:41:32] go.mod: adding generated comments
+++ [0425 17:41:33] vendor: updating internal modules
+++ [0425 17:41:33] vendor: running 'go mod vendor'
+++ [0425 17:41:34] vendor: updating vendor/LICENSES
+++ [0425 17:41:39] vendor: creating OWNERS file
+++ [0425 17:41:39] NOTE: don't forget to handle vendor/* files that were added or removed
➜  kubernetes git:(improve-mount-detection) ./hack/lint-dependencies.sh          
Check all unwanted dependencies and update its status.
All pinned versions of checked dependencies match their preferred version.
➜  kubernetes git:(improve-mount-detection) git status
On branch improve-mount-detection
Your branch is up to date with 'origin/improve-mount-detection'.

nothing to commit, working tree clean
➜  kubernetes git:(improve-mount-detection) git log --oneline -n 3                                       
32fe234004b (HEAD -> improve-mount-detection, origin/improve-mount-detection) Fix type in mount_windows.go
0097b228bff Use faster mountdetections using openat2 for IsNotMountPoint
c625fb38ac1 Update mountinfo to v0.6.1.
➜  kubernetes git:(improve-mount-detection) 
  kubernetes git:(improve-mount-detection) ./hack/pin-dependency.sh github.com/moby/sys/mountinfo v0.6.1
Running: go mod download github.com/moby/sys/mountinfo@v0.6.1
Resolved to github.com/moby/sys/mountinfo@v0.6.1
Running: go mod edit -require github.com/moby/sys/mountinfo@v0.6.1
Running: go mod edit -replace github.com/moby/sys/mountinfo=github.com/moby/sys/mountinfo@v0.6.1

Run hack/update-vendor.sh to rebuild the vendor directory
➜  kubernetes git:(improve-mount-detection) ✗ ./hack/update-vendor.sh                                      
+++ [0425 17:40:58] logfile at /tmp/update-vendor.QxrQ/update-vendor.log
+++ [0425 17:40:58] go.mod: update staging references
+++ [0425 17:41:03] go.mod: propagate to staging modules
+++ [0425 17:41:03] go.mod: sorting staging modules
+++ [0425 17:41:18] go.mod: tidying
+++ [0425 17:41:32] go.mod: adding generated comments
+++ [0425 17:41:33] vendor: updating internal modules
+++ [0425 17:41:33] vendor: running 'go mod vendor'
+++ [0425 17:41:34] vendor: updating vendor/LICENSES
+++ [0425 17:41:39] vendor: creating OWNERS file
+++ [0425 17:41:39] NOTE: don't forget to handle vendor/* files that were added or removed
➜  kubernetes git:(improve-mount-detection) ./hack/lint-dependencies.sh          
Check all unwanted dependencies and update its status.
All pinned versions of checked dependencies match their preferred version.
➜  kubernetes git:(improve-mount-detection) git status
On branch improve-mount-detection
Your branch is up to date with 'origin/improve-mount-detection'.

nothing to commit, working tree clean
➜  kubernetes git:(improve-mount-detection) git log --oneline -n 3                                       
32fe234004b (HEAD -> improve-mount-detection, origin/improve-mount-detection) Fix type in mount_windows.go
0097b228bff Use faster mountdetections using openat2 for IsNotMountPoint
c625fb38ac1 Update mountinfo to v0.6.1.
➜  kubernetes git:(improve-mount-detection) 
  kubernetes git:(improve-mount-detection) ./hack/pin-dependency.sh github.com/moby/sys/mountinfo v0.6.1
Running: go mod download github.com/moby/sys/mountinfo@v0.6.1
Resolved to github.com/moby/sys/mountinfo@v0.6.1
Running: go mod edit -require github.com/moby/sys/mountinfo@v0.6.1
Running: go mod edit -replace github.com/moby/sys/mountinfo=github.com/moby/sys/mountinfo@v0.6.1

Run hack/update-vendor.sh to rebuild the vendor directory
➜  kubernetes git:(improve-mount-detection) ✗ ./hack/update-vendor.sh                                      
+++ [0425 17:40:58] logfile at /tmp/update-vendor.QxrQ/update-vendor.log
+++ [0425 17:40:58] go.mod: update staging references
+++ [0425 17:41:03] go.mod: propagate to staging modules
+++ [0425 17:41:03] go.mod: sorting staging modules
+++ [0425 17:41:18] go.mod: tidying
+++ [0425 17:41:32] go.mod: adding generated comments
+++ [0425 17:41:33] vendor: updating internal modules
+++ [0425 17:41:33] vendor: running 'go mod vendor'
+++ [0425 17:41:34] vendor: updating vendor/LICENSES
+++ [0425 17:41:39] vendor: creating OWNERS file
+++ [0425 17:41:39] NOTE: don't forget to handle vendor/* files that were added or removed
➜  kubernetes git:(improve-mount-detection) ./hack/lint-dependencies.sh          
Check all unwanted dependencies and update its status.
All pinned versions of checked dependencies match their preferred version.
➜  kubernetes git:(improve-mount-detection) git status
On branch improve-mount-detection
Your branch is up to date with 'origin/improve-mount-detection'.

nothing to commit, working tree clean
➜  kubernetes git:(improve-mount-detection) git log --oneline -n 3                                       
32fe234004b (HEAD -> improve-mount-detection, origin/improve-mount-detection) Fix type in mount_windows.go
0097b228bff Use faster mountdetections using openat2 for IsNotMountPoint
c625fb38ac1 Update mountinfo to v0.6.1.
➜  kubernetes git:(improve-mount-detection) 
```

I can see no changes.

Reply:
Hmm! not sure what happened; doing this from master gave the output that you gave. I will resolve this.